chatkit-core/src/main/kotlin/com/pusher/chatkit/CurrentUser.kt

```diff
@@ -233,45 +348,90 @@ class CurrentUser(
             callback: (Result<Int, Error>) -> Unit
     ) = makeCallback({ syncCurrentUser.sendMessage(roomId, messageText, attachment) }, callback)
 
+    /**
+     * Send a [messageText] to the [roomId]. The [callback] will be called with a [Result] when the
+     * operation is complete. If the operation was successful you will receive an [Int] which
+     * contains the message id, or an [Error] informing you of what went wrong.
+     */
     fun sendSimpleMessage(
             roomId: String,
             messageText: String,
             callback: (Result<Int, Error>) -> Unit
     ) = makeCallback({ syncCurrentUser.sendSimpleMessage(roomId, messageText) }, callback)
 
+    /**
+     * @see [sendSimpleMessage]
+     */
     fun sendSimpleMessage(
             room: Room,
             messageText: String,
             callback: (Result<Int, Error>) -> Unit
     ) = makeCallback({ syncCurrentUser.sendSimpleMessage(room, messageText) }, callback)
 
+    /**
+     * Sends some message [parts] to the [room]. For more information on what the [parts] can be check
+     * out the [docs][https://pusher.com/docs/chatkit/reference/android].
+     * The [callback] will be called with a [Result] when the operation is complete. If the
+     * operation was successful you will receive an [Int] which contains the message id, or an
+     * [Error] informing you of what went wrong.
+     */
     fun sendMultipartMessage(
             room: Room,
             parts: List<NewPart>,
             callback: (Result<Int, Error>) -> Unit
     ) = makeCallback({ syncCurrentUser.sendMultipartMessage(room, parts) }, callback)
 
+    /**
+     * @see [sendMultipartMessage]
+     */
     fun sendMultipartMessage(
             roomId: String,
             parts: List<NewPart>,
             callback: (Result<Int, Error>) -> Unit
     ) = makeCallback({ syncCurrentUser.sendMultipartMessage(roomId, parts) }, callback)
 
+    /**
+     * Sets that the current user is typing in the [room]. You can send as many of these as you like,
+     * the SDK will handle rate limiting the requests. The [callback] will be called with a [Result]
+     * when the operation is complete. If the operation was successful you will receive a [Unit], or
+     * an [Error] informing you of what went wrong.
+     */
```

Comment:
It doesn't really make sense to say that you receive a `Unit`, it's a type representing nothing. Perhaps we could phrase it more like "There is no value associated with the success case", or "In the case of success there is no value to receive".

Reply:
I'm going to update all the other comments to match this too